front-end/src/components/CreateEditPostModal.tsx

```diff
@@ -141,7 +141,7 @@ export default function CreateEditPostModal(props: Props){
     } else if (res.status >= 200) {
       const post:Post = res.data;
       if(isCreate){
-        if(props.prependToFeed !== undefined){
+        if(props.prependToFeed !== undefined && post.visibility === "Public"){
```

Comment:
you can use the static variable instead of "Public". i think its Post.visibility.Public